From dd0c7ee51727f94a4508ff03562e272854b0c5e9 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Fri, 12 Oct 2012 20:30:24 -0500
Subject: [PATCH 051/254] arm: socfpga: Enable USB for socfpga

Upstream: git://git.rocketboards.org/linux-socfpga.git

-Add USB entries to device tree source.
-Add USB to socfpga_defconfig

Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
(cherry picked from commit a84ef32df11fcd2db331dbda64cb8ba2f162fe80)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/socfpga.dtsi     |   36 ++++++++++++++++++++++++++++--------
 arch/arm/configs/socfpga_defconfig |    6 ++++++
 2 files changed, 34 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index 00b3a76..bb7c9a2 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -175,14 +175,34 @@
 			reg-io-width = <4>;
 		};
 
-		rstmgr@ffd05000 {
-				compatible = "altr,rst-mgr";
-				reg = <0xffd05000 0x1000>;
-			};
+		usb0: usb@ffb00000 {
+				compatible = "snps,dwc-otg";
+				reg = <0xffb00000 0xffff>;
+				interrupts = <0 125 4>;
+				dma-mask = <0xffffffff>;
+				ulpi-ddr = <0>;
+				host-rx-fifo-size = <512>;
+				dev-rx-fifo-size = <512>;
+				dev-nperio-tx-fifo-size = <4096>;
+				dev-perio-tx-fifo-size = <512 512 512 512 512 512
+					512 512 512 512 512 512 512 512 512>;
+				dev-tx-fifo-size = <512 512 512 512 512 512
+					512 512 512 512 512 512 512 512 512>;
+		};
 
-		sysmgr@ffd08000 {
-				compatible = "altr,sys-mgr";
-				reg = <0xffd08000 0x4000>;
-			};
+		usb1: usb@ffb40000 {
+				compatible = "snps,dwc-otg";
+				reg = <0xffb40000 0xffff>;
+				interrupts = <0 128 4>;
+				dma-mask = <0xffffffff>;
+				ulpi-ddr = <1>;
+				host-rx-fifo-size = <512>;
+				dev-rx-fifo-size = <512>;
+				dev-nperio-tx-fifo-size = <4096>;
+				dev-perio-tx-fifo-size = <512 512 512 512 512 512
+					512 512 512 512 512 512 512 512 512>;
+				dev-tx-fifo-size = <512 512 512 512 512 512
+					512 512 512 512 512 512 512 512 512>;
+		};
 	};
 };
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 22c96e7..ab43ee7 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -62,6 +62,12 @@ CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_NR_UARTS=2
 CONFIG_SERIAL_8250_RUNTIME_UARTS=2
 CONFIG_SERIAL_8250_DW=y
+CONFIG_USB=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_DWC_HDRC=y
+CONFIG_USB_MASS_STORAGE=y
+CONFIG_DWC_OTG_MODE=y
 CONFIG_MMC=y
 CONFIG_MMC_DW=y
 CONFIG_MMC_DW_IDMAC=y
-- 
1.7.5.4

